internal/data/resource_repository.go

```diff
@@ -241,3 +248,85 @@ func (r *resourceRepository) GetDB() *gorm.DB {
 func (r *resourceRepository) GetTransactionManager() usecase.TransactionManager {
 	return r.transactionManager
 }
+
+func (r *resourceRepository) FindCommonRepresentationsByVersion(tx *gorm.DB, key bizmodel.ReporterResourceKey, currentVersion uint) ([]CommonRepresentationsByVersion, error) {
+	var results []CommonRepresentationsByVersion
+
+	// Use provided transaction or fall back to regular DB session
+	db := tx
+	if db == nil {
+		db = r.db.Session(&gorm.Session{})
+	}
+
+	query := db.Table("reporter_resources rr").
+		Select("cr.data, cr.version").
+		Joins("JOIN common_representations cr ON rr.resource_id = cr.resource_id").
+		Where("LOWER(rr.local_resource_id) = LOWER(?)", key.LocalResourceId().Serialize()).
```

Comment:
I am considering moving these where clauses to a common helper function in the repository, so we can use them in both the lookup functions